main.py

```diff
@@ -1,93 +1,120 @@
-from digitalio import DigitalInOut, Direction, Pull
-import board
-from light import LightStrip
-import patterns
-from robot_communication import recorded_mode, get_mode, get_states
-import time
-
-Colors = {
-    "RED": (255, 0, 0),
-    "ORANGE": (255, 100, 100),
-    "GRAY": (100, 100, 100),
-    "GREEN": (0,255,0),
-    "BLUE":(0,0,255)}
-
-robot_state: list[int]
-mainloop_count = 0
+"""
+MODES:
+[31] No code on robot:
+RoboRIO inactive/not receiving serial data from RoboRIO
 
+[32] Disabled without auto
+RoboRIO is disabled, and has no autonomous program set
 
-def match_state_with_pattern(light: LightStrip):
-    if light.state == 1:
-        light.primary = Colors["BLUE"]
-        light.pattern = patterns.static
+[33] Disabled with auto
+RoboRIO is disabled, abd dose have autonomous program set
 
-    elif light.state == 2:
-        light.primary = Colors["GREEN"]
-        light.pattern = patterns.static
 
-    elif light.state == 3:
-        light.primary = Colors["GRAY"]
-        light.pattern = patterns.static
+--< Modes for when RoboRIO is enabled >--
+Priority: 1 > 2 > 3 > 4
 
-    elif light.state == 4:
-        light.primary = Colors["RED"]
-        light.pattern = patterns.flashing
+[1] Signal:
+For communication with teammates to activate amplifier
 
-    elif light.state == 5:
-        light.primary = Colors["ORANGE"]
-        light.pattern = patterns.static
-
-    elif light.state == 6:
-        light.primary = Colors["BLUE"]
-        light.secondary = Colors["GRAY"]
-        light.pattern = patterns.railgun
+[2] Ready to shoot:
+The robot shooter and arm is in position, ready to shoot note
 
+[3] Has note:
+The robot currently is holding a note
 
-def main_loop(lights: list[LightStrip]):
-    # Assuming we are getting a state
-    # Also, "action" or "rainbow" states should have higher priority than static
-    global robot_state
-    global recorded_mode
-    patterns.loopcount += 1
+[4] Vision:
+The robot spots a note (tells the driver that they can press a button for
+robot to retrieve a note)
+"""
 
-    new_states = get_states()
-    for light in lights:
-        if new_states[light.channel] != robot_state[light.channel]:
-            print("CHanged")
-            light.pattern_starting_loop = patterns.loopcount
 
-    robot_state = new_states # Placed here because we always want to update our states
-    recorded_mode = get_mode()
+import supervisor
+import board
+from lightstrip import LightStrip
+import patterns
+import time
+import sys
+
+# Serial data variables
+serial = sys.stdin
+tolerance = 2
+tolerance_count = 0
+current_mode = "31"
+last_mode = "31"
+
+# Create light objects
+strip1 = LightStrip(board.D5, 8)
+lightstrips: tuple[LightStrip] = (strip1,)
+
+
+# Retrieves and returns serial data sent by Java side
+def get_serial_data() -> str:
+    # Return serial data if any
+    if supervisor.serial_bytes_available > 0:
```

Comment:
nit: in Python, if a value is nonzero and you evaluate it in an `if` statement, it will evaluate to true, so the `> 0` part should be unnecessary.